src/ess/loki/workflow.py

```diff
@@ -1,24 +1,139 @@
 # This file is used by beamlime to create a workflow for the Loki instrument.
-# The function `live_workflow` is registered as the entry point for the workflow.
+# The callable `live_workflow` is registered as the entry point for the workflow.
+import sciline
 import scipp as sc
-from beamlime import WorkflowResult
+import scippnexus as snx
 from scippneutron.io.nexus.load_nexus import JSONGroup
 
+from ess.loki.general import (
+    get_monitor_data,
+    get_source_position,
+    monitor_to_tof,
+    patch_monitor_data,
+)
+from ess.loki.io import load_nexus_monitor, load_nexus_source
+from ess.sans.conversions import monitor_to_wavelength, sans_monitor
+from ess.sans.types import (
+    Filename,
+    FilePath,
+    Incident,
+    IofQ,
+    MonitorType,
+    NeXusMonitorName,
+    RunType,
+    SampleRun,
+    ScatteringRunType,
+    Transmission,
+    WavelengthBins,
+    WavelengthMonitor,
+)
 
-def live_workflow(group: JSONGroup) -> WorkflowResult:
-    """Example live workflow function for Loki.
 
-    This function is used to process data with beamlime workflows.
-    """
-    # We do not consume the incoming data right now.
-    _ = group
-    return {
-        'IofQ': sc.DataArray(
-            data=sc.zeros(dims=['Q'], shape=[100]),
+class MonitorHistogram(
+    sciline.ScopeTwoParams[RunType, MonitorType, sc.DataArray], sc.DataArray
+):
+    ...
+
+
+class IofQHistogram(sciline.Scope[ScatteringRunType, sc.DataArray], sc.DataArray):
+    ...
+
+
+def _hist_monitor_wavelength(
+    wavelength_bin: WavelengthBins, monitor: WavelengthMonitor[RunType, MonitorType]
+) -> MonitorHistogram[RunType, MonitorType]:
+    return monitor.hist(wavelength=wavelength_bin)
+
+
+def _empty_iofq(_: ScatteringRunType) -> IofQ[ScatteringRunType]:
+    return IofQ[ScatteringRunType](
+        sc.DataArray(
+            data=sc.zeros(dims=["Q"], shape=[100]),
             coords={
-                'Q': sc.linspace(
-                    dim='Q', start=0.01, stop=0.3, num=101, unit='1/angstrom'
+                "Q": sc.linspace(
+                    dim="Q", start=0.01, stop=0.3, num=101, unit="1/angstrom"
                 )
             },
         )
-    }
+    )
+
+
+def _hist_iofq(i_of_q: IofQ[ScatteringRunType]) -> IofQHistogram[ScatteringRunType]:
+    return IofQHistogram[ScatteringRunType](i_of_q.hist("Q"))
```

Comment:
`hist` does not accept string argument, I think. Why is this code part of this PR?